pkg/dynamic/discovery/discovery.go

```diff
@@ -198,6 +198,16 @@ func (rm *ResourceMap) HasSynced() bool {
 	return rm.groupVersions != nil
 }
 
+func (rm *ResourceMap) WaitForSync() {
```

Comment:
I know if it mostly not present in the code, but if we wish to make it an API I would ask to add documentation there in form in godoc comments on public methods